src/main/java/ladder/domain/Ladder.java

```diff
@@ -0,0 +1,74 @@
+package ladder.domain;
+
+import ladder.rowgenerator.RowGenerator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Ladder implements Iterable<Row> {
+  private final List<Row> rows;
+
+  public static Ladder from(final int[][] ladder) {
+    return new Ladder(Arrays.stream(ladder)
+            .map(Row::of)
+            .collect(Collectors.toList()));
+  }
+
+  public static Ladder from(final List<Row> rows) {
+    return new Ladder(rows);
+  }
+
+  public static Ladder of(final Integer height, final Integer width, final RowGenerator generator) {
+    final List<Row> ladder = new ArrayList<>(height);
+
+    IntStream.range(0, height)
+            .forEach(i -> ladder.add(Row.fromGenerator(generator, width)));
+
+    return new Ladder(ladder);
+  }
+
+  private Ladder(final List<Row> rows) {
+    if (rows.isEmpty()) {
+      throw new IllegalArgumentException("잘못된 사다리 입력입니다.");
+    }
+
+    Integer width = rows.get(0).size();
+    for (Row row : rows) {
+      if (row.isNotSizeOf(width)) {
+        throw new IllegalArgumentException("길이가 다른 행(Row) 입력이 있습니다.");
+      }
+    }
```

Comment:
Stream API를 활용하거나 함수로 분리해줘도 좋을 것 같네요!
```suggestion
    boolean hasInvalidRow = rows.anyMathch( ... );
    if(hasINvalidRow) { ... }
```

Reply:
메서드로 분리하였습니다 :)